src/bootstrap/native.rs

```diff
@@ -178,6 +177,11 @@ impl Step for Llvm {
             .define("LLVM_TARGET_ARCH", target_native.split('-').next().unwrap())
             .define("LLVM_DEFAULT_TARGET_TRIPLE", target_native);
 
+        if !self.emscripten {
+            let polly_src = builder.src.join("src/polly");
```

Comment:
Polly is included in LLVM checkout, why don't just use it?

Reply:
This was a minimal merge of the old patch into the current master. I already got rid of the Polly submodule, but I tried to not change the code to keep some separation between the updated base and the actual semantic changes that are needed to make this work.
It will use the LLVM submodule's Polly.